deno_syntax.py

```diff
@@ -0,0 +1,26 @@
+import os
+import re
+import sublime
+import sublime_plugin
+
+
+RE_DENO_DEPS_CACHES_DIRPATH = re.compile(r".+\/deno\/deps\/.+", re.IGNORECASE)
+
+
+class DenoSyntaxListener(sublime_plugin.EventListener):
+    def on_load_async(self, view: sublime.View):
+        self.check_deno_syntax(view)
+
+    def check_deno_syntax(self, view: sublime.View):
+        if not (file_name := view.file_name()):
+            return
+        if not (view.scope_name(0).startswith("text.plain")):
+            return
+        if not RE_DENO_DEPS_CACHES_DIRPATH.match(file_name):
+            return
+        self.set_deno_syntax(view, "d." + os.path.basename(file_name[0:-60]) + ".ts")
+
+    def set_deno_syntax(self, view: sublime.View, name: str):
+        view.set_read_only(True)
+        view.set_name(name)
+        view.assign_syntax(sublime.find_syntax_for_file(name))
```

Comment:
- The `:=` syntax is new in Python 3.8 so it wouldn't work in this package as this package runs in 3.3 runtime.
- I'd probably just check the currently set syntax using `view.settings().get('syntax')` rather than syntax at position 0.
- I think such script would make sense to exist separately or as part of some generic Deno package as one might want that behavior regardless if using LSP-typescript or not. And those not using Deno should not need it also. So I must say that I'm not very keen on including it here. :/